### ISSUE TITLE
Auto-sync VSCT and Auto-sync to Resx are not working in VS2017 - No error, nothing happens.

### ISSUE THREAD

**Issue by kdawg1406** [madskristensen/ExtensibilityTools#54]
### Installed product versions
- Visual Studio:  2017 Enterprise
- This extension: 1.10.188

### Description
After right clicking the vsixmanifest and selecting Auto-sync to Resx and Icon Files, nothing happens.  No error and no files are created.

### Steps to recreate
1. Open VS 2017
2. Create an extension
3. Right click the vsixmanifest file and select Auto-sync to Resx and Icon Files

### Current behavior
Nothing happens.

### Expected behavior
Files would be created under the vsixmanifestfile.

**Comment by kdawg1406** [comment]
If I manually run the custom tool for the manifest file I get this error: 

Cannot find custom tool 'VsixManifestGenerator' on this system.

**Comment by kdawg1406** [comment]
If I manually run the custom tool for the .vsct file I get this error:
Cannot find custom tool 'VsctGenerator' on this system.

**Comment by kdawg1406** [comment]
None of these registry entries are in my registry.  What can I do?

[$RootKey$\InstalledProducts\ExtensibilityToolsPackage]
@="#110"
"Package"="{f8330d54-0469-43a7-8fc0-7f19febeb897}"
"PID"="1.10.188"
"ProductDetails"="#112"
"LogoID"="#400"
[$RootKey$\Packages\{f8330d54-0469-43a7-8fc0-7f19febeb897}]
@="ExtensibilityToolsPackage"
"InprocServer32"="$WinDir$\SYSTEM32\MSCOREE.DLL"
"Class"="MadsKristensen.ExtensibilityTools.ExtensibilityToolsPackage"
"CodeBase"="$PackageFolder$\ExtensibilityTools.dll"
[$RootKey$\AutoLoadPackages\{f1536ef8-92ec-443c-9ed7-fdadf150da82}]
"{f8330d54-0469-43a7-8fc0-7f19febeb897}"=dword:00000000
[$RootKey$\Menus]
"{f8330d54-0469-43a7-8fc0-7f19febeb897}"=", Menus.ctmenu, 1"
[$RootKey$\ToolsOptionsPages\Extensibility Tools]
@="#101"
"Package"="{f8330d54-0469-43a7-8fc0-7f19febeb897}"
[$RootKey$\ToolsOptionsPages\Extensibility Tools\General]
@="#102"
"Package"="{f8330d54-0469-43a7-8fc0-7f19febeb897}"
"Page"="{d72dc55f-84b0-31b4-883f-ec5085fcf50c}"
"Keywords"="pkgdef;vsct"
[$RootKey$\AutomationProperties\Extensibility Tools]
@="#101"
"Package"="{f8330d54-0469-43a7-8fc0-7f19febeb897}"
[$RootKey$\AutomationProperties\Extensibility Tools\General]
@="##102"
"Package"="{f8330d54-0469-43a7-8fc0-7f19febeb897}"
[$RootKey$\AutomationProperties\Extensibility Tools\General]
"Name"="Extensibility Tools.General"
[$RootKey$\ToolWindows\{a4acbaf7-9223-4fbe-8177-38a8cf4ab59a}]
@="{f8330d54-0469-43a7-8fc0-7f19febeb897}"
"Name"="MadsKristensen.ExtensibilityTools.ThemeColorsToolWindow.SwatchesWindow"
[$RootKey$\Generators]
[$RootKey$\Generators\{164b10b9-b200-11d0-8c61-00a0c91e29d5}]
[$RootKey$\Generators\{164b10b9-b200-11d0-8c61-00a0c91e29d5}\VsctGenerator]
@="Generates .NET source code for given VS IDE GUI definitions."
"CLSID"="{a6a34300-fa6b-4f86-a8ba-e1fea8d24922}"
"GeneratesDesignTimeSource"=dword:00000001
[$RootKey$\CLSID]
[$RootKey$\CLSID\{a6a34300-fa6b-4f86-a8ba-e1fea8d24922}]
@="Generates .NET source code for given VS IDE GUI definitions."
"Class"="MadsKristensen.ExtensibilityTools.VSCT.Generator.VsctCodeGenerator"
"InprocServer32"="$WinDir$\SYSTEM32\MSCOREE.DLL"
"ThreadingModel"="Both"
"CodeBase"="$PackageFolder$\ExtensibilityTools.dll"
[$RootKey$\Generators]
[$RootKey$\Generators\{fae04ec1-301f-11d3-bf4b-00c04f79efbc}]
[$RootKey$\Generators\{fae04ec1-301f-11d3-bf4b-00c04f79efbc}\VsctGenerator]
@="Generates .NET source code for given VS IDE GUI definitions."
"CLSID"="{a6a34300-fa6b-4f86-a8ba-e1fea8d24922}"
"GeneratesDesignTimeSource"=dword:00000001
[$RootKey$\CLSID]
[$RootKey$\CLSID\{a6a34300-fa6b-4f86-a8ba-e1fea8d24922}]
@="Generates .NET source code for given VS IDE GUI definitions."
"Class"="MadsKristensen.ExtensibilityTools.VSCT.Generator.VsctCodeGenerator"
"InprocServer32"="$WinDir$\SYSTEM32\MSCOREE.DLL"
"ThreadingModel"="Both"
"CodeBase"="$PackageFolder$\ExtensibilityTools.dll"
[$RootKey$\Generators]
[$RootKey$\Generators\{fae04ec1-301f-11d3-bf4b-00c04f79efbc}]
[$RootKey$\Generators\{fae04ec1-301f-11d3-bf4b-00c04f79efbc}\VsixManifestGenerator]
@="Automatically generates the .resx file based on the .vsixmanifest file values"
"CLSID"="{f86b5aa5-733c-4e8f-8d3b-ea6e9b97b343}"
"GeneratesDesignTimeSource"=dword:00000001
[$RootKey$\CLSID]
[$RootKey$\CLSID\{f86b5aa5-733c-4e8f-8d3b-ea6e9b97b343}]
@="Automatically generates the .resx file based on the .vsixmanifest file values"
"Class"="MadsKristensen.ExtensibilityTools.VsixManifest.ResxFileGenerator"
"InprocServer32"="$WinDir$\SYSTEM32\MSCOREE.DLL"
"ThreadingModel"="Both"
"CodeBase"="$PackageFolder$\ExtensibilityTools.dll"

**Comment by kdawg1406** [comment]
Solved the problem by download the source, change project to target 4.6.1 and updated the references that were broken.  Rebuilt, works perfect.

**Comment by midiway** [comment]
Well, what about sharing it and making a PR?

**Comment by kdawg1406** [comment]
I would be MUCH better if @madskristensen performed the work and posted it.  He has so many customers across many versions of Visual Studio, me uploading what I did would be reckless.

What I did took 10 minutes.

Download Source
Restore NuGet Packages
Updated project to target .NET 4.6.1
Fixed all references, many were not found on "my machine" not sure why.
Set Embed Interrop tyepes to false for envdte and evndte80
I also changed the product id so that it would not interferer with Mad's real version.
That's it.

Karl
